grpc/diatheke.proto

```diff
@@ -281,10 +306,40 @@ message PushTextRequest {
   string text = 2;
 }
 
+// Request to change the current story of a session.
+message StoryRequest {
+  // ID of the session that will have its story changed.
+  string session_id = 1;
+
+  // ID of the story to switch to. This ID is defined by the model
+  // used to create the session.
+  string story_id = 2;
+
+  // Parameters to set in session memory before executing the specified
+  // story. Some stories in the model may make assumptions about which
+  // parameters have already been defined, so it is important to be
+  // familiar with the model requirements for any given story.
+  map<string, string> parameters = 3;
+
+  // If true, the given story will not be executed until the session
+  // completes the current stories and returns back to the main story.
+  // If false, the current story in the session will be immediately
+  // interrupted to execute the specified story.
+  bool wait_for_start = 4;
+
+  // If true, once the given story has finished, Diatheke will return
+  // the session to the place in the model where it was when this request
+  // was received. This is useful when the change in story represents a
```

Comment:
Should the integrator be thinking of this as a stack then?  We push a new thing on, run it, and then pop it off, returning to the previous state?